lib/Runtime/Types/DictionaryTypeHandler.cpp

```diff
@@ -2742,7 +2742,7 @@ namespace Js
             //
 
             Js::PropertyId pid = iter.CurrentKey()->GetPropertyId();
-            if(!DynamicTypeHandler::ShouldMarkPropertyId_TTD(pid) | (!descriptor.IsInitialized) | (descriptor.Attributes & PropertyDeleted))
+            if(!DynamicTypeHandler::ShouldMarkPropertyId_TTD(pid) || (!descriptor.IsInitialized) || (descriptor.Attributes & PropertyDeleted))
```

Comment:
@mrkmarron Had some rationale for using `|` instead of `||`. I wonder if he has any comments on this change.